src/irmin-pack/unix/file_manager.ml

```diff
@@ -528,10 +586,9 @@ struct
     let make_lower () =
       match lower_root with
       | None -> Ok None
-      | Some path ->
-          let+ l = Lower.v ~readonly:false ~volume_num path in
-          (* TODO: if [volume_num] is 0, we need to migrate
-             (or return error if [no_migrate = true]) *)
```

Comment:
Just want to comment on this TODO that I left. I thought this would be the spot for the migration (to mirror what happens in `create_rw`), but I see why you put the migration code in the path of loading the payload. :+1: